client/internal/clientcommon.go

```diff
@@ -186,6 +288,7 @@ func (c *ClientCommon) StartConnectAndRun(runner func(ctx context.Context)) {
 		return
 	}
 	c.runCancel = runCancel
+	c.isStarted = true
```

Comment:
This was a bug that resulted in racing with the goroutine started below. If the goroutine was quick enough we would get a callback while isStarted was still false.